ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -162,33 +171,32 @@ dataFiles =
 
 includesDependencies :: GhcFlavor -> [FilePath]
 includesDependencies ghcFlavor =
-   map (root ++ )
-   [ "ghcautoconf.h"
-   , "ghcplatform.h"
-   , "ghcversion.h"]
+  map (root </> ) [ "ghcautoconf.h", "ghcplatform.h", "ghcversion.h" ]
    where
-      root = (case ghcFlavor of
-                GhcMaster -> stage0LibPath
-                Ghc8101 -> stage0LibPath
-                _ -> "ghc-lib/generated") ++ "/"
+      root =
+        case ghcFlavor of
+          GhcMaster -> stage0Lib
+          Ghc8101 -> stage0Lib
+          _ -> ghcLibGeneratedPath
 
 derivedConstantsDependencies :: GhcFlavor -> [FilePath]
 derivedConstantsDependencies ghcFlavor =
-   map (root ++)
-   [ "DerivedConstants.h"
-   , "GHCConstantsHaskellExports.hs"
-   , "GHCConstantsHaskellType.hs"
-   , "GHCConstantsHaskellWrappers.hs"
-   ]
+   map (root </>)
```

Comment:
Do you really want `(</>)` here?  This ends up in the cabal file so your cabal file is now platform dependent.

Reply:
`import System.FilePath.Posix((</>)) -- Make sure we generate / on all platforms.` probably means it's safe but I take your point. i'll fix that.

Reply:
ah didn’t realize that. Then it’s fine :+1: